app/code/Piimega/Maksuturva/Model/Gateway/Implementation.php

```diff
@@ -62,6 +63,7 @@ function __construct(
         $this->_taxHelper = $taxHelper;
         $this->_calculationModel = $calculationModel;
         $this->_maksuturvaForm = $maksuturvaForm;
+        $this->eventManager = $eventManager ?: \Magento\Framework\App\ObjectManager::getInstance()->get(\Magento\Framework\Event\ManagerInterface::class);
```

Comment:
instead of using ObjectManager it's better to add `\Magento\Framework\Event\ManagerInterface` as `use` and pass it to constructor.

Reply:
Can change. This was done because to get backwards compatibility (in case someone has extended this class).